src/components/ui/SimpleDropdown.tsx

```diff
@@ -4,50 +4,52 @@ import { CaretDownIcon } from '@patternfly/react-icons';
 
 type SimpleDropdownProps = {
   current?: string;
-  values: string[];
+  defaultValue: string;
+  items: { [key: string]: string };
   setValue: (value?: string) => void;
   isDisabled: boolean;
 };
 
 export const SimpleDropdown: React.FC<SimpleDropdownProps> = ({
   current,
-  values,
+  defaultValue,
+  items,
   setValue,
   isDisabled,
 }) => {
   const [isOpen, setOpen] = React.useState(false);
-  const items = values.map((value) => (
-    <DropdownItem key={value} id={value}>
-      {value}
+  const dropdownItems = Object.keys(items).map((key) => (
+    <DropdownItem key={key} id={key}>
+      {items[key]}
     </DropdownItem>
   ));
 
-  const onRoleSelect = React.useCallback(
```

Comment:
Renamed a few things as is was specific to Role, but the dropdown is supposed to be generic